src/crystal/system/event_loop/file_descriptor.cr

```diff
@@ -0,0 +1,16 @@
+abstract class Crystal::EventLoop
+  module FileDescriptor
+    # Reads at least one byte from the file descriptor into *slice* and continues
+    # fiber when the read is complete.
+    # Returns the number of bytes read.
+    abstract def read(file_descriptor : Crystal::System::FileDescriptor, slice : Bytes) : Int32
+
+    # Writes at least one byte from *slice* to the file descriptor and continues
+    # fiber when the write is complete.
+    # Returns the number of bytes written.
+    abstract def write(file_descriptor : Crystal::System::FileDescriptor, slice : Bytes) : Int32
```

Comment:
**suggestion**: same and "when the write is complete" can be ambiguous (_writes fully?_)

> Writes at least one byte from *slice* to the file descriptor.
> Blocks the current fiber if the file descriptor isn't ready for writing, otherwise returns immediately.
> Returns the number of bytes written (up to *slice.size*).